fedimint-core/src/util/error.rs

```diff
@@ -0,0 +1,130 @@
+use std::fmt::Formatter;
+use std::{error, fmt};
+
+/// A wrapper with `fmt::Display` for any `E : Error` that will print chain
+/// of causes
+pub struct FmtErrorCompact<'e, E>(pub &'e E);
+
+impl<E> fmt::Display for FmtErrorCompact<'_, E>
+where
+    E: error::Error,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut source_opt: Option<&(dyn std::error::Error)> = Some(self.0);
+
+        while source_opt.is_some() {
+            let source = source_opt.take().expect("Just checked");
```

Comment:
Could this be a `while let Some(source) = source_opt.take() {`? `is_some`+`expect` sounds wrong.